license-bill-of-materials.go

```diff
@@ -301,7 +307,7 @@ func getPackagesInfo(gopath string, pkgs []string) ([]*PkgInfo, error) {
 	decoder := json.NewDecoder(bytes.NewBuffer(out))
 	for _, pkg := range pkgs {
 		info := &PkgInfo{}
-		err := decoder.Decode(info)
+		err = decoder.Decode(info)
```

Comment:
It is generally a good practice to have a separate variable here since difference functions can return different type of errors.
Another way to handle it is to declare the err variable at the beginning of the function along with the type and then reuse that through out the method.